### ISSUE TITLE
Add download meters

### ISSUE THREAD

**Issue by NoozAbooz** [piware14/pi-ware#37]
What does this suggestion do to pi-ware?
It helps people gauge what's popular and what do download, as well as telling app contributors if their app is popular.

This could be implemented similarly to RPI Imager's telemetry, where installing a app sends a signal to a website to increase download counts in a JSON file for a app, and remove counts upon uninstall.

**Comment by Painadath** [comment]
please give a completely description

**Comment by Painadath** [comment]
@mobilegmYT

**Comment by NoozAbooz** [comment]
Wdym?

**Comment by IsaacMvmv** [comment]
@mobilegmYT Like a download count of each app?

**Comment by NoozAbooz** [comment]
> @mobilegmYT Like a download count of each app?

Yep

**Comment by oxmc** [comment]
I think we can implement that.

> What does this suggestion do to pi-ware?
> It helps people gauge what's popular and what do download, as well as telling app contributors if their app is popular.
> 
> This could be implemented similarly to RPI Imager's telemetry, where installing a app sends a signal to a website to increase download counts in a JSON file for a app, and remove counts upon uninstall.

**Comment by IsaacMvmv** [comment]
> I think we can implement that.
> 
> > What does this suggestion do to pi-ware?
> > It helps people gauge what's popular and what do download, as well as telling app contributors if their app is popular.
> > This could be implemented similarly to RPI Imager's telemetry, where installing a app sends a signal to a website to increase download counts in a JSON file for a app, and remove counts upon uninstall.

BuT, that number could be manipulated, also I can make an script to change that number to whatever we want, taking the line of the installer and modifying it :/

**Comment by oxmc** [comment]
oh, no. We will have the file download each time.

**Comment by IsaacMvmv** [comment]
k

**Comment by ZaydTheCoder** [comment]
There can be a .txt file that stores the data. We would need it not to be in a obvious directory so users don't modify it.

**Comment by IsaacMvmv** [comment]
But script says where can be located, and we can view the script

**Comment by ZaydTheCoder** [comment]
Yes, there is no way to hide where it stores the file

**Comment by IsaacMvmv** [comment]
I suggest ip saving, like:

IP=$(curl ifconfig.me)
if $IP is same as saved somewhere, not send 2 website

**Comment by oxmc** [comment]
> There can be a .txt file that stores the data. We would need it not to be in a obvious directory so users don't modify it.

We can also change view permissons

**Comment by ZaydTheCoder** [comment]
Yes, we can but we would have to do it on the install aswell because otherwise, they will be able to see in the install where does it store the file, even if it hides it.

**Comment by IsaacMvmv** [comment]
Or chmod 700 the file
Much more we cant do

**Comment by oxmc** [comment]
Hmmm, this requires research.

**Comment by ZaydTheCoder** [comment]
Yes, it does

**Comment by ZaydTheCoder** [comment]
Would it be worth trying to hide the downloads data? What I mean is that is it worth the effort to hide the file? A user would have to burrow through pi-ware files to find and even if they did find it, they would just be able to change what the download data is.

**Comment by oxmc** [comment]
Yeah, if they modifiy it, it gets replaced when pi-ware auto-updates.

**Comment by oxmc** [comment]
> Would it be worth trying to hide the downloads data? What I mean is that is it worth the effort to hide the file? A user would have to burrow through pi-ware files to find and even if they did find it, they would just be able to change what the download data is.

And i codename every file per it's purpose, so... yeah.

**Comment by oxmc** [comment]
> Would it be worth trying to hide the downloads data? What I mean is that is it worth the effort to hide the file? A user would have to burrow through pi-ware files to find and even if they did find it, they would just be able to change what the download data is.

And i codename every file per it's purpose, so... yeah.

**Comment by ZaydTheCoder** [comment]
Did you type the same message twice?

**Comment by oxmc** [comment]
no, it does this sometimes.

**Comment by IsaacMvmv** [comment]
Some good things

**Comment by oxmc** [comment]
> Some good things
?

**Comment by IsaacMvmv** [comment]
Yes

**Comment by gl91306** [comment]
The best way to do this I think is to make an api request to pi wares site each time you download an app and each time you check the download meters where the info will be stored and can be accessed. The api request would have to have a key or and id to make sure the request is coming form pi-ware and not spammed.

**Comment by Painadath** [comment]
@gl91306  or uploading or updating a text when user clicks but IDK how to do this

**Comment by Painadath** [comment]
may be this https://roytuts.com/python-flask-file-upload-example/

**Comment by oxmc** [comment]
already found a way.

**Comment by oxmc** [comment]
http://pi-ware-telemetry.ml/main/apps/

**Comment by Painadath** [comment]
> http://pi-ware-telementry.ml/main/apps/

but how does it work?

**Comment by oxmc** [comment]
You send a POST request to the site.

**Comment by oxmc** [comment]
In bash.

**Comment by Painadath** [comment]
it may worl

**Comment by Painadath** [comment]
work

**Comment by oxmc** [comment]
it does, you want to test it?

**Comment by Painadath** [comment]
> it does, you want to test it?

give me the code in discord DM

**Comment by oxmc** [comment]
k

**Comment by oxmc** [comment]
So this has been figured out, now it just needs to be implemented.

**Comment by oxmc** [comment]
Progress has been made, we are now adding it to pi-ware.

**Comment by ZaydTheCoder** [comment]
Has this been implemented @oxmc ?

**Comment by oxmc** [comment]
> Has this been implemented @oxmc ?

The script has been made, we just need to make the store make a json file.

**Comment by gl91306** [comment]
yayy :D

**Comment by confuzled** [comment]
Yaay:D

**Comment by ZaydTheCoder** [comment]
Umm, @oxmc does this work yet?

**Comment by oxmc** [comment]
Not yet, still adding it @ZaydTheCoder

**Comment by ZaydTheCoder** [comment]
k

**Comment by oxmc** [comment]
Ok, implemented, but need to make store upload it.

**Comment by confuzled** [comment]
any updates?

**Comment by oxmc** [comment]
None so far.

**Comment by confuzled** [comment]
k

**Comment by oxmc** [comment]
@mobilegmYT @IsaacMvmv @Painadath @gl91306

I have the script working, i just need to have the store send run the file.

**Comment by Painadath** [comment]
but uploading in python is a pain i think

**Comment by Painadath** [comment]
but uploading in python is a pain i think

**Comment by CleanMachine1** [comment]
Why would you add telemetry? It doesn't add anything, especially if the telemetry is IP. Its just spyware.

**Comment by gl91306** [comment]
I mean its in pi-apps and its a way to know which apps are the most used and stuff. And it doesn't track IP.

**Comment by gl91306** [comment]
and you have the option to opt out of telemetry on first startup

**Comment by CleanMachine1** [comment]
Oh ok, sorry I was ignorant, just saw this thread and wanted to give an outsiders opinion.

If it is opt **IN** then it is fine.

but if it is like VSCode where you have to opt out then :-1:

**Comment by gl91306** [comment]
when you first install pi-ware it asks you: Do you wish to allow your data to be sent to allow analytics, with a yes or no option. You get to choose at startup.

**Comment by confuzled** [comment]
But wouldn't that make analytics less accurate?

**Comment by oxmc** [comment]
Yes, but I believe it's only fair if a user has the choice to send data.

**Comment by CleanMachine1** [comment]
> But wouldn't that make analytics less accurate?

opt out practice to me is against open source code.
Some analytics do not outway the privacy of people.

**Comment by confuzled** [comment]
i feel like i probably should have added more context to that message ._. 


what i was more meaning was that with analyitics, having an option kind of ruins the whole point by reducing the accuracy , but having forced analytics in open source is really dumb.

i dont know what i was trying to say in my original message but  ¯\_(ツ)_/¯

**Comment by dimaguy** [comment]
Don't forget to take into account that GDPR is a thing, and you'd have to store personal data(IPs) in europe and stuff
